junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -826,6 +826,8 @@ private static void findNestedClasses(Class<?> clazz, Set<Class<?>> candidates)
 				.collect(toList());
 		// @formatter:on
 
+		localMethods.sort(ReflectionUtils::defaultMethodSorter);
+
```

Comment:
We could move this up to where `localMethods` is computed an use the Stream API: `.sorted(ReflectionUtils::defaultMethodSorter).collect(...)`.